Feature: add docs for handling merge conflicts in the IDE (#320)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud/cloud-ide/handling-merge-conflicts.md

---
title: "Handling Merge Conflicts"
id: "handling-merge-conflicts"
---

> Merge conflicts often occur when multiple users are concurrently making edits to the same section in the same file. This makes it difficult for Git to determine the appropriate change to keep. The merge conflict process then gives the users the ability to sort out which lines of code should be kept and committed. Here we'll show you how you'd resolve merge conflicts the Cloud IDE!

Reviewer: ```suggestion
Merge conflicts often occur when multiple users are concurrently making edits to the same section in the same file. This makes it difficult for Git to determine the appropriate change to keep. The merge conflict process gives users the ability to sort out which lines of code should be kept and committed. Here we'll show you how you'd resolve merge conflicts in the IDE!
```

## Running into a merge conflict

In this example we have a column that represents the total number of orders your customer has ever had. It's currently named `number_of_orders` but your stakeholder feels like this could named better. 

> You and your teammate come in to make the change but decide to go two separate routes. You rename this column to `total_number_of_orders` and your teammate decides to name it `historical_order_count`. 

Reviewer: ```suggestion
You and a teammate make changes to the column name, but have gone two separate routes. You rename this column to `total_number_of_orders` and your teammate has renamed it `historical_order_count`. 
```

> Your teammate got to make their change first so you encounter the merge conflict when you press commit.

Reviewer: ```suggestion
Your teammate has committed their change first, so you encounter a merge conflict when you press commit.
```
